spinalcordtoolbox/vertebrae/detect_c2.py

```diff
@@ -0,0 +1,149 @@
+        description="tools to detect C2/C3 intervertebral disc with countception deep learning network ",
+        epilog="EXAMPLES:\n",
+        add_help=None,
+        formatter_class=SmartFormatter,
+        prog=os.path.basename(__file__).strip('.py'))
+
+    mandatoryArguments = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatoryArguments.add_argument(
+        '-i',
+        required=True,
+        help="Input image. Example: t2.nii.gz",
+        metavar=Metavar.file,
+    )
+    mandatoryArguments.add_argument(
+        '-c',
+        required=True,
+        help="contrast",
+    )
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        '-h',
+        '--help',
+        action='help',
+        help="Show this help message and exit")
+    optional.add_argument(
+        '-o',
+        help="output name",
+        metavar=Metavar.str,
+    )
+
+    optional.add_argument(
+        '-net',
+        help="Network to use",
+        default='CC',
+    )
+
+    return parser
+
+
+def main(args=None):
+    """
+    Main function
+    :param args:
+    :return:
+    """
+    # get parser args
+    if args is None:
+        args = None if sys.argv[1:] else ['--help']
+    parser = get_parser()
+    arguments = parser.parse_args(args=args)
+    Im_input = Image(arguments.i)
+    contrast = arguments.c
+
+    global cuda_available
+    cuda_available = torch.cuda.is_available()
+
+    if arguments.net == 'CC':
+        model = ModelCountception_v2(inplanes=1, outplanes=1)
+
+        if contrast == 't1':
+            model.load_state_dict(torch.load(
+                os.path.join(sct.__sct_dir__, 'spinalcordtoolbox/vertebrae/checkpoints/Countception_floatC2T2.model'),
+                map_location='cpu')['model_weights'])
+
+        elif contrast == 't2':
+            model.load_state_dict(torch.load(
+                os.path.join(sct.__sct_dir__,'spinalcordtoolbox/vertebrae/checkpoints/Countception_floatC2T2.model'),
+                map_location='cpu')['model_weights'])
+
+    elif arguments.net == 'AttU':
+        model = AttU_Net()
+        if contrast == 't1':
+            model.load_state_dict(torch.load(
+                '~/luroub_local/lurou_local/deep_VL_2019/ivado_med/scripts_vertebral_labeling/checkpoints/attunet_c2T1.model',
+                map_location='cpu')['model_weights'])
+
+        elif contrast == 't2':
+            model.load_state_dict(torch.load(
+                '/home/GRAMES.POLYMTL.CA/luroub/luroub_local/lurou_local/deep_VL_2019/ivado_med/scripts_vertebral_labeling/checkpoints/attunet_c2T2.model',
+                map_location='cpu')['model_weights'])
+
+
+
+
+    else:
+        sct.printv('Error...unknown contrast. please select between t2 and t1.')
+        return 100
+
+    if cuda_available:
+        model = model.cuda()
+    model = model.float()
+
+    sct.printv('retrieving input...')
+    Im_input.change_orientation('RPI')
```

Comment:
this is axial orientation-- i thought detection was on sagittal plane? maybe you want to check, and also comment to explain this line of code